### ISSUE TITLE
Lambda expressions with attributes are formatted incorrectly

### ISSUE THREAD

**Issue by PathogenDavid** [dotnet/roslyn#56543]
**Version Used**: Visual Studio 2022 Preview 4, Compiler version: '4.0.0-4.21458.2 (2bfff7b9)'. Language version: 10.0.

**Steps to Reproduce**:

1. In a new .NET 6 console project, populate `Program.cs` with the following:

```csharp
var a = [A] short () => 100;
var b = [A] () => 100;
var c = short () => 100;

public sealed class AAttribute : Attribute { }
```

2. Format the document using Edit -> Advanced -> Format Document or run Code Cleanup

**Expected Behavior**:

No changes, the code is well-formatted.

**Actual Behavior**:

The space between the assignment operator and the attribute is removed:

```csharp
var a =[A] short () => 100;
var b =[A] () => 100;
var c = short () => 100;

public sealed class AAttribute : Attribute { }
```

----------------

As far as I know there is not an editorconfig setting for the spaces around the assignment operator, but to remove any doubt I also reproduced this with with [Roslyn's `.editorconfig`](https://github.com/dotnet/roslyn/blob/2ff43addcf69c5cbf2e3756daf551976e9d464b6/.editorconfig).

Relates to test plan https://github.com/dotnet/roslyn/issues/52192

**Comment by jcouv** [comment]
Fix merged for 17.1

**Comment by PathogenDavid** [comment]
Thanks, @jcouv!
